### PR TITLE
Add support of the packet length and several protocols filter

### PR DESCRIPTION
**Pull request by sbarguil**
Add support of the packet length and several protocols filter. 
Add ICMP field in IPv4 and IPv6 packets

### CODE REVIEW

**Comment by sbarguil** [comment]
Cames from https://github.com/openconfig/models/pull/988

File: release/models/acl/openconfig-packet-match.yang [hunk 1]
```diff
@@ -432,4 +454,55 @@ module openconfig-packet-match {
...
+      description
+        "Also known as control messages.";
+      reference
+        "RFC 792: Internet Control Message Protocol
+         RFC 4443: Internet Control Message Protocol (ICMPv6)
+                   for Internet Protocol Version 6 (IPv6)
+                   Specification.";
+    }
+    leaf icmp-code {
+      type uint8;
```
Comment (robshakir) [inline release/models/acl/openconfig-packet-match.yang:497]: Should we have this as an `enumeration`? We have an internal model that has the definitions of the ICMPv4 and ICMPv6 types that I can propose as an addition if this makes sense.

**Review by robshakir — CHANGES_REQUESTED** [review]
Per the usual requests -- can we add some references to >= 2 implementations that support this behaviour? I believe that they exist. I'd just like to make sure that we have this clearly documented for each PR.

File: release/models/acl/openconfig-packet-match.yang [hunk 2]
```diff
@@ -27,11 +27,17 @@ module openconfig-packet-match {
     field is omitted from a match expression, the effect is a
     wildcard ('any') for that field.";
 
-  oc-ext:openconfig-version "1.3.1";
+  oc-ext:openconfig-version "1.4.0";
+
+  revision "2022-03-29" {
+    description
+      "Add ICPM Fields for filtering.";
```
Comment (sourcequench) [inline release/models/acl/openconfig-packet-match.yang:34]: s/ICPM/ICMP

File: release/models/acl/openconfig-packet-match.yang [hunk 3]
```diff
@@ -432,4 +454,55 @@ module openconfig-packet-match {
...
+        uses ip-icmp-header-fields-config;
+      }
+    }
+  }
+
+  grouping ip-icmp-header-fields-config {
+    description
+      "Collection of ICMP header fields that can be
+       used to set up a match filter.";
+    leaf icmp-type {
```
[the remark this answers is not in the document]
Comment (robshakir) [reply release/models/acl/openconfig-packet-match.yang:486]: Addressed in the forked PR.

**Pull request closed without merging**